File: Winter/std::sort and std::set are better anway/3.12 homework/insertion.cpp
#include <iostream>
#include <vector>
#include <cstdlib> 
#include <time.h> 
using namespace std;
void insertion_sort(vector<int> &list){
    for (int i=0; i<(list.size()-1); i++){
        int index = i + 1;
        while ((list[index] < list[index - 1]) && (index > 0)){
            int temp = list[index];
            list[index] = list[index - 1];
            list[index - 1] = temp; 
            index--;
        }
    }
}
void print_list(vector<int> list){
    cout << "list is: ";
    for (auto i: list){
        cout << i << " ";
    }
    cout << endl;
}
int main(){
    srand(time(0)); 
    vector<int> list = {};
    for (int i=0; i<(rand() % 100); i++){
        list.push_back(rand() % 100);
    }
    cout << "Unsorted "; print_list(list);
    insertion_sort(list);
    cout << "Sorted "; print_list(list);
}

File: Winter/std::sort and std::set are better anway/3.12 homework/insertion.py
import random
mylist = list({})
for i in range(10):
    mylist.append(random.randint(1,10000))
def insertion_sort(Alist):
    List = Alist
    for i in range(len(List) - 1):
        index = i + 1
        while ((List[index] < List[index - 1]) and (index > 0)):
            temp = List[index]
            List[index] = List[index - 1]
            List[index - 1] = temp
            index -= 1
    return List
print("Unsorted list: ", mylist)
print("Sorted List: ", insertion_sort(mylist))